PermissionsTestApp/GrantedPermissionsViewController.m

```diff
@@ -0,0 +1,99 @@
+//
+//  GrantedPermissionsViewController.m
+//  ISHPermissionKit
+//
+//  Created by Felix Lamouroux on 10.06.16.
+//  Copyright © 2016 iosphere GmbH. All rights reserved.
+//
+
+#import "GrantedPermissionsViewController.h"
+#import "AppDelegate.h"
+#import <ISHPermissionKit/ISHPermissionRequest+All.h>
+
+NSString * const GrantedPermissionsViewControllerCell = @"cell";
+
+@interface GrantedPermissionsViewController ()
+@property NSArray *permissionsGranted;
+@property NSArray *permissionsNotGranted;
+@end
+
+@implementation GrantedPermissionsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:GrantedPermissionsViewControllerCell];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    [self reloadPermissions];
+}
+
+- (void)viewDidLayoutSubviews {
+    [super viewDidLayoutSubviews];
+    self.tableView.contentInset = UIEdgeInsetsMake(self.topLayoutGuide.length, 0, self.bottomLayoutGuide.length, 0);
+}
+
+- (void)reloadPermissions {
+    NSSet *permissions = [NSSet setWithArray:[AppDelegate requiredPermissions]];
+    NSSet *grantedPermissions = [ISHPermissionRequest grantedPermissionsForCategories:permissions];
+
+    self.permissionsGranted = [[grantedPermissions allObjects] sortedArrayUsingSelector:@selector(compare:)];
+    NSMutableSet *missingPermissions = [permissions mutableCopy];
+    [missingPermissions minusSet:grantedPermissions];
+    self.permissionsNotGranted = [[missingPermissions allObjects] sortedArrayUsingSelector:@selector(compare:)];
+    [self.tableView reloadData];
+}
+
+- (NSArray *)permissionsForSection:(NSInteger)section {
+    if (section) {
+        return self.permissionsGranted;
+    }
+
+    return self.permissionsNotGranted;
+}
+
+- (NSNumber *)permissionAtIndexPath:(NSIndexPath *)indexPath {
+    NSArray *permissions = [self permissionsForSection:indexPath.section];
+
+    if (permissions.count <= indexPath.row) {
```

Comment:
Off by one: (count = 5), (row = 5) => Check passes > Boom!

If think `indexPath.row >= permissions.count` is easier to read.